File: LCS/LCS.java
package LCS;
import java.util.Scanner;
import java.lang.Math;

public class LCS {
	public static int[][] mat;
	// as a class, both the input strings should also be its member variable
	
	public static int lengthOfLCS(String s1, String s2) {
		int str1L = s1.length();
		int str2L = s2.length();
		mat = new int[str1L+1][str2L+1];
		for(int i = 0; i<=str1L; i++) {
			mat[i][0] = 0;
		}
		for(int i = 0; i<=str2L; i++) {
			mat[0][i] = 0;
		}
		for(int i = 1; i<=str1L; i++) {
			for(int j = 1;j<=str2L; j++) {
//				if the character matches at this place pick the diagonal and increment by 1
				if(s1.charAt(i-1) == s2.charAt(j-1)) {
					mat[i][j] = mat[i-1][j-1]+1;
				}
//				else pic the max of left one or the above
				else {
					mat[i][j] = Math.max(mat[i-1][j], mat[i][j-1]);
				}
			}
		}
		return mat[str1L][str2L];
	}
	
	public static String stringOfLCS(String s1, String s2) {
		// as we have made changes, 
		// we can be trace them in last row or column of the main mat
		String res = "";
		int i = s1.length();
		int j = s2.length();
		while(mat[i][j] != 0) {
			if(mat[i][j] != mat[i-1][j]) {
				res += s1.charAt(i-1);
			}
			i--;
		}
		String Temp = "";
		for(int k = res.length()-1; k>=0 ;k--) {
			Temp += res.charAt(k);
		}
		return Temp;
	}

	public static void main(String[] args) {
		// TODO Auto-generated method stub
		Scanner scan = new Scanner(System.in);
		String str1 = scan.nextLine();
		String str2 = scan.nextLine();
		System.out.println("Length of Longest Common Subsequence: "+lengthOfLCS(str1, str2));
		System.out.println("Longest Common Subsequence is: '"+stringOfLCS(str1, str2)+"'");		
		scan.close();
	}
}

File: LCS/LCS.go
package main

import (
	"fmt"
	"math"
	"strings"
)

func lcsss(X string, Y string, m int, n int) int {
	if m == 0 || n == 0 {
		return 0
	}
	if X[m-1] == Y[n-1] {
		return 1 + lcsss(X, Y, m-1, n-1)
	}
	return int(math.Max(float64(lcsss(X, Y, m, n-1)), float64(lcsss(X, Y, m-1, n))))

}

func main() {
	var str1, str2 string
	fmt.Println("Input first String: ")
	fmt.Scanf("%s", &str1)
	fmt.Println("\nInput second String:  ")
	fmt.Scanf("%s", &str2)
	m := len(strings.Split(str1, ""))
	n := len(strings.Split(str2, ""))
	res := lcsss(str1, str2, m, n)
	fmt.Println("\nLength of LCS is", res)
}
